include/k4arecord/record.hpp

```diff
@@ -67,13 +67,41 @@ class record
         return m_handle != nullptr;
     }
 
+    /** Returns true if two recorder objects refer to the same k4a_record_t, false otherwise
+     */
+    bool operator==(const record &other) const noexcept
+    {
+        return m_handle == other.m_handle;
+    }
+
+    /** Returns false if the record object is valid, true otherwise
+     */
+    bool operator==(std::nullptr_t) const noexcept
+    {
+        return m_handle == nullptr;
+    }
+
+    /** Returns true if two record objects wrap different k4a_record_t instances, false otherwise
+     */
+    bool operator!=(const record &other) const noexcept
+    {
+        return m_handle != other.m_handle;
+    }
+
+    /** Returns true if the record object is valid, false otherwise
+     */
+    bool operator!=(std::nullptr_t) const noexcept
+    {
+        return m_handle != nullptr;
+    }
```

Comment:
I'm not sure these work with the semantics of k4a::record - it's not a pseudo-pointer type like k4a::image, we delete the copy constructor so two records should never be equal and we don't allow assigning nullptr to the playback, so allowing comparison to nullptr seems really weird

Reply:
It is also common in C to check handles against NULL, so this seems fine to me